include/mbedtls/aes.h

```diff
@@ -329,6 +329,57 @@ int mbedtls_aes_crypt_ctr( mbedtls_aes_context *ctx,
                        unsigned char *output );
 #endif /* MBEDTLS_CIPHER_MODE_CTR */
 
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+/**
+ * \brief      This function performs an AES-XTS encryption or decryption
+ *             operation for an entire XTS sector.
+ *
+ *             The key for tweak computation is separate from the key for AES
+ *             block encryption. So, two AES contexts must be supplied. XXX
+ *             Design review: It's a bit strange to have two AES ctxs.
+ *             Alternatives include a new key/ctx initialization function that
+ *             can handle the extra key needed by XTS, so long as that fits in
+ *             the existing context. This is sometimes handled as one big key
+ *             instead of two separate ones, key1 concat key2. Could store a
+ *             pointer in the ctx for the tweak's ctx, or could make a new ctx
+ *             for xts operations.
+ *
+ *             AES-XTS encrypts or decrypts blocks based on their location as
```

Comment:
The interface in the standard allows random access to a single AES blocks inside of a data unit. It seems to me that this can be useful depending on the hardware and the application.

Do you think it would worth adding this capability to the API?

Reply:
I considered this. Supporting sub-data-unit operations complicates both the API and the implementation. XTS is typically used with whole data units, where the data unit is the size of the block device block. There may be some optimization for small reads to be gained if we support sub-data-unit access, as the entire device block doesn't need decrypting.

If we need sub-data-unit operations, I would propose we add it with a new API at a later time, and keep this simpler API around both as a backwards compatibility wrapper and as an easier to use version of the API.